### PR TITLE
Remove concept of temporary dynamic IL

### PR DESCRIPTION
**Pull request by davidwrighton**
- This was once used for profiler IL rewrite scenarios, but hasn't been in use for some time

### CODE REVIEW

File: src/coreclr/vm/ceeload.cpp [hunk 1]
```diff
@@ -1288,20 +1276,9 @@ TADDR Module::GetDynamicIL(mdToken token, BOOL fAllowTemporary)
     CrstHolder ch(m_debuggerSpecificData.m_pDynamicILCrst);
 #endif
 
-    // Both hash tables are lazily allocated, so if they're NULL
+    // The hash tables is lazily allocated, so if it is NULL
```
Comment (jkotas) [inline src/coreclr/vm/ceeload.cpp:1279]:
```suggestion
    // The hash table is lazily allocated, so if it is NULL
```

**Review by jkotas — APPROVED** [review]
Thanks

**Review by davmason — APPROVED** [review]
LGTM as is, but also we could take it a step further and get rid of the whole dynamic IL concept by using code versioning and merging it with ReJIT logic.

**Comment by davidwrighton** [comment]
@davmason I'm looking into that as well, but the first thing was to get rid of this code that was somewhat confusing me, and had no purpose at all.

**Pull request merged**